core/src/main/scala/org/apache/spark/scheduler/JobWaiter.scala

```diff
@@ -17,6 +17,10 @@
 
 package org.apache.spark.scheduler
 
+import sun.misc.{Signal, SignalHandler}
```

Comment:
I don't really think there is an easy way around the ```sun.misc.Signal/SignalHander``` approach. 

I have to admit that my concern is moot if all major implementations support this (can anyone confirm this?). I just like the JLine approach better because this doesn't cause Compile/Run-time errors when classes cannot be found, it will just default to the current behavior.